src/measure_theory/integral/circle_transform.lean

```diff
@@ -174,4 +173,308 @@ begin
   exact this,
 end
 
+/--Cauchy integral form of a function at `z` in a disk of radius `R`-/
+def circle_integral_form [complete_space E] (R : ℝ) (z : ℂ) (f : ℂ → E) : (ℂ → E) :=
+  λ w, (2 * π * I : ℂ)⁻¹ • (∮ z in C(z, R), (z - w)⁻¹ • f z)
+
+lemma circle_intgral_form_eq_int [complete_space E] (R : ℝ) (z : ℂ) (f : ℂ → E) :
+  circle_integral_form R z f = λ w,
+ ∫ (θ : ℝ) in 0..2 * π, (circle_transform R z w f) θ :=
+begin
+    simp_rw [circle_transform, circle_integral_form, circle_integral,
+      interval_integral.integral_smul],
+end
+
+lemma circle_transform_circle_int [complete_space E] (R : ℝ) (z w : ℂ) (f : ℂ → E) :
+  ∫ (θ : ℝ) in 0..2 * π, circle_transform R z w f θ =
+ (2 * π * I : ℂ)⁻¹ • ∮ z in C(z, R), (z - w)⁻¹ • f z :=
+begin
+  simp_rw [circle_transform, circle_integral,deriv_circle_map, circle_map],
+  simp only [real_smul, nsmul_eq_mul, nat.cast_bit0, nat.cast_one, one_div,
+  interval_integral.integral_smul, zero_add],
+end
+
+lemma circle_transform_has_deriv_at (R : ℝ) (z : ℂ) (f : ℂ → ℂ) :
+  ∀ (t : ℝ) y ∈ ball z R, has_deriv_at (λ y, (circle_transform R z y f) t)
+  ((circle_transform_deriv R z y f) t) y :=
+begin
+  intros y x hx,
+  simp only [circle_transform, circle_transform_deriv, algebra.id.smul_eq_mul,
+  ←mul_assoc, deriv_circle_map],
+  apply_rules [has_deriv_at.mul_const, has_deriv_at.const_mul],
+  have H : has_deriv_at (λ (y_1 : ℂ), (circle_map z R y - y_1)) (-1 ) x,
+  by {apply has_deriv_at.const_sub, apply has_deriv_at_id,},
+  have hfin := has_deriv_at.inv H (sub_ne_zero.2 (circle_map_ne_mem_ball hx y)),
+  simp only [one_div, neg_neg] at hfin,
+  apply hfin,
+end
+
+lemma circle_transform_ae_measurable {R : ℝ} {f : ℂ → ℂ} (hR : 0 < R)
+  (z x : ℂ) (hx : x ∈ ball z R) (hf : continuous_on f (sphere z R)) :
+  ∀ᶠ y in 𝓝 x, ae_measurable (( λ w, (λ θ, (circle_transform R z w f θ))) y)
+  (volume.restrict (Ι 0 (2 * π))):=
+begin
+  rw filter.eventually_iff_exists_mem,
+  obtain ⟨ε', He, HB⟩ := (exists_ball_subset_ball hx),
+  refine  ⟨(ball x ε'), _⟩,
+  simp only [metric.ball_mem_nhds x He, exists_true_left],
+  intros y hy,
+  apply continuous_on.ae_measurable ((continuous_circle_transform hR hf (HB hy))).continuous_on
+    (measurable_set_interval_oc),
+end
+
+lemma circle_interval_integrable {R : ℝ} {f : ℂ → ℂ} (hR : 0 < R)
+ (z x : ℂ) (hx : x ∈ ball z R) (hf : continuous_on f (sphere z R)) :
+ interval_integrable ((λ w, (λ θ, (circle_transform R z w f θ))) x) volume 0 (2 * π) :=
+begin
+  apply (continuous_on.interval_integrable),
+  apply (continuous_circle_transform hR hf hx).continuous_on,
+end
+
+lemma circle_transform_deriv_ae_measurable {R : ℝ} (hR : 0 < R)
+  (z x : ℂ) (hx : x ∈ ball z R) (f : ℂ → ℂ) (hf : continuous_on f (sphere z R)) :
+   ae_measurable (( λ w, (λ θ, (circle_transform_deriv R z w f θ))) x)
+  (volume.restrict (Ι 0 (2 * π))) :=
+begin
+ apply continuous_on.ae_measurable ((continuous_circle_transform_deriv hR hf (hx))).continuous_on
+    (measurable_set_interval_oc),
+end
```

Comment:
```suggestion
begin
 apply continuous_on.ae_measurable ((continuous_circle_transform_deriv hR hf (hx))).continuous_on
    (measurable_set_interval_oc),
end
```
as above this can be golfed into a term using `show_term`